### PR TITLE
soplex: init at 7.1.0

### PR DESCRIPTION
**Pull request by david-r-cox**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by david-r-cox** [comment]
Result of `nixpkgs-review pr 273296` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>soplex</li>
  </ul>
</details>

**Comment by david-r-cox** [comment]
Result of `nixpkgs-review pr 273296` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>soplex</li>
  </ul>
</details>

File: pkgs/by-name/so/soplex/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "soplex";
+  version = "6.0.4";
+
+  src = fetchFromGitHub {
+    owner = "scipopt";
+    repo = "soplex";
+    rev = "release-604";
```
Comment (AndersonTorres) [inline pkgs/by-name/so/soplex/package.nix:14]:
```suggestion
    rev = "release-${builtins.replaceStrings ["."] [""] version}";
```

File: pkgs/by-name/so/soplex/package.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  doCheck = true;
+
+  meta = with lib; {
+    description = "Sequential object-oriented simPlex";
+    homepage = "https://scipopt.org";
+    license = with licenses; [ asl20 ];
+    maintainers = with maintainers; [ david-r-cox ];
+    mainProgram = "soplex";
+  };
```
Comment (AndersonTorres) [inline pkgs/by-name/so/soplex/package.nix:30]:
```suggestion
  meta = {
    homepage = "https://scipopt.org";
    description = "Sequential object-oriented simPlex";
    license = with lib.licenses; [ asl20 ];
    mainProgram = "soplex";
    maintainers = with lib.maintainers; [ david-r-cox ];
    platforms = lib.platforms.unix;
  };
```

File: pkgs/by-name/so/soplex/package.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    owner = "scipopt";
+    repo = "soplex";
+    rev = "release-604";
+    hash = "sha256-miTgfouMk+TofHO6aJVY6op9JM5k1ekLkkGAfu3QVsM=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
```
Comment (AndersonTorres) [inline pkgs/by-name/so/soplex/package.nix:21]:
```suggestion

  strictDeps = true;

```

File: pkgs/by-name/so/soplex/package.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{ lib
+, stdenv
+, cmake
+, fetchFromGitHub
+}:
+
+stdenv.mkDerivation rec {
```
Comment (AndersonTorres) [inline pkgs/by-name/so/soplex/package.nix:7]: Have you cogitated `finalAttrs` design pattern?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/so/soplex/package.nix (hunk 4 above)
Comment (david-r-cox) [reply pkgs/by-name/so/soplex/package.nix]: TIL -- thanks!

**Comment by eclairevoyant** [comment]
You can run `nix-shell` then `nixfmt` the file to pass CI

**Comment by david-r-cox** [comment]
@NixOS/nixpkgs-merge-bot merge

**Review by AndersonTorres — APPROVED** [review]

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**